File: src/lua/generic/mathBenchmark.lua
MathBenchmark = class(MathBase, function(a, writer, printToConsole)
    MathBase.init(a, writer, printToConsole)
    a.name = "MathBenchmark"
end)

function MathBenchmark:benchImplementation()
    local R = 0
    local li = 0.0
    for i=1,self.iterrations do
        R = R + self:doMath(i, li)
    end
    return R
end

File: src/js/mathBenchmark.js
var MathBenchmark = /** @class */ (function (_super) {
	__extends(MathBenchmark, _super);

	function MathBenchmark(writer) {
		var _this = _super.call(this, writer) || this;
		return _this;
	}

	MathBenchmark.prototype.BenchImplementation = function () {
		_super.prototype.BenchImplementation.call(this);
		var R = 0;
		var li = 0;
		for (var i = 0; i < this.Iterrations; li = i, i++) {
			this.R += _super.DoMath(i, li);
		}
		return R;
	};

	return MathBenchmark;
}(MathBase));

if (typeof module !== 'undefined' && module.exports) {
	module.exports = {
		MathBenchmark: MathBenchmark
	};
}